src/bindings/c/ov/tests/ov_c_api_test.cpp

```diff
@@ -0,0 +1,1718 @@
+
+#include <gtest/gtest.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <opencv2/opencv.hpp>
+#include <condition_variable>
+#include <mutex>
+#include "test_model_repo.hpp"
+#include <fstream>
+
+#include "c_api/ov_c_api.h"
+#include "openvino/openvino.hpp"
+
+std::string xml_std = TestDataHelpers::generate_model_path("test_model", "test_model_fp32.xml"),
+            bin_std = TestDataHelpers::generate_model_path("test_model", "test_model_fp32.bin"),
+            input_image_std = TestDataHelpers::generate_image_path("224x224", "dog.bmp"),
+            input_image_nv12_std = TestDataHelpers::generate_image_path("224x224", "dog6.yuv");
+
+const char* xml = xml_std.c_str();
+const char* bin = bin_std.c_str();
+const char* input_image = input_image_std.c_str();
+const char* input_image_nv12 = input_image_nv12_std.c_str();
+
+std::mutex m;
+bool ready = false;
+std::condition_variable condVar;
+#ifdef _WIN32
+    #ifdef __MINGW32__
+        std::string plugins_xml_std = TestDataHelpers::generate_ieclass_xml_path("plugins_mingw.xml");
+    #else
+        std::string plugins_xml_std = TestDataHelpers::generate_ieclass_xml_path("plugins_win.xml");
+    #endif
+#elif defined __APPLE__
+        std::string plugins_xml_std = TestDataHelpers::generate_ieclass_xml_path("plugins_apple.xml");
+#else
+        std::string plugins_xml_std = TestDataHelpers::generate_ieclass_xml_path("plugins.xml");
+#endif
+const char* plugins_xml = plugins_xml_std.c_str();
+
+#define OV_EXPECT_OK(...) EXPECT_EQ(ov_status_e::OK, __VA_ARGS__)
+#define OV_ASSERT_OK(...) ASSERT_EQ(ov_status_e::OK, __VA_ARGS__)
+#define OV_EXPECT_NOT_OK(...) EXPECT_NE(ov_status_e::OK, __VA_ARGS__)
+#define OV_EXPECT_ARREQ(arr1, arr2) EXPECT_TRUE(std::equal(std::begin(arr1), std::end(arr1), std::begin(arr2)))
+
+std::map<ov_element_type_e, size_t> element_type_size_map = {
+        {ov_element_type_e::BOOLEAN, 8},
+        {ov_element_type_e::BF16, 16},
+        {ov_element_type_e::F16, 16},
+        {ov_element_type_e::F32, 32},
+        {ov_element_type_e::F64, 64},
+        {ov_element_type_e::I4, 4},
+        {ov_element_type_e::I8, 8},
+        {ov_element_type_e::I16, 16},
+        {ov_element_type_e::I32, 32},
+        {ov_element_type_e::I64, 64},
+        {ov_element_type_e::U1, 1},
+        {ov_element_type_e::U4, 4},
+        {ov_element_type_e::U8, 8},
+        {ov_element_type_e::U16, 16},
+        {ov_element_type_e::U32, 32},
+        {ov_element_type_e::U64, 64}};
+#define GET_ELEMENT_TYPE_SIZE(a) element_type_size_map[a]
+
+size_t read_image_from_file(const char* img_path, unsigned char *img_data, size_t size) {
+    FILE *fp = fopen(img_path, "rb+");
+    size_t read_size = 0;
+
+    if (fp) {
+        fseek(fp, 0, SEEK_END);
+        if (ftell(fp) >= size) {
+            fseek(fp, 0, SEEK_SET);
+            read_size = fread(img_data, 1, size, fp);
+        }
+        fclose(fp);
+    }
+    return read_size;
+}
+
+void mat_2_tensor(const cv::Mat& img, ov_tensor_t* tensor)
```

Comment:
Looks like we don't use this function

Reply:
Maybe it is a function in early version of this PR, we forget remove it, thanks your finding!